File: Top 100 Codes/01-Getting Started/25-Automorphic number/Automorphicnumber.py
# Take user input for the number
num = int(input("Enter a number: "))

# Convert the number to a string
num_str = str(num)

# Calculate the square of the number and convert it to a string
square = num ** 2
square_str = str(square)

# Check if the square ends with the original number
if square_str.endswith(num_str):
    print(f"{num} is an Automorphic Number (Square: {square})")
else:
    print(f"{num} is not an Automorphic Number (Square: {square})")

File: Top 100 Codes/01-Getting Started/25-Automorphic number/Automorphicnumber.cpp
#include <iostream>
using namespace std;

// Function to check if a number is Automorphic
int isAutomorphic(int n) {
    int square = n * n; // Calculate the square of the number

    while (n != 0) {
        // Check the last digit of n and square
        if (n % 10 != square % 10) {
            return 0; // Not an Automorphic number
        }

        // Remove the last digit from both n and square
        n /= 10;
        square /= 10;
    }

    // If the loop completes, the number is Automorphic
    return 1;
}

int main() {
    int n;

    // Take user input for the number
    cout << "Enter a number: ";
    cin >> n;

    int square = n * n; // Calculate the square of the input number

    // Check if the number is Automorphic and print the result
    if (isAutomorphic(n)) {
        cout << "Num: " << n << ", Square: " << square << " - is Automorphic" << endl;
    } else {
        cout << "Num: " << n << ", Square: " << square << " - is not Automorphic" << endl;
    }

    return 0;
}
